rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -60,9 +67,17 @@ public T doExecute(Object... args) throws Throwable {
         String selectPKSQL = buildSelectSQL(paramAppenderList);
         try {
             if (originalAutoCommit) {
+                /*
+                 * prevent release local lock before get global lock
+                 * dose not need the savepoint if original auto commit was true
+                 */
                 conn.setAutoCommit(false);
+            } else if (dbmd.supportsSavepoints()) {
+                // need the savepoint if original auto commit was false
```

Comment:
In order to release the local db lock when global lock conflict, create a save point if original auto commit was false, then use the save point here to release db lock if necessary during global lock checking?

Reply:
maybe like this? I change some sentence order.

In order to release the local db lock when global lock conflict, create a save point if original auto commit was false, then use the save point here to release db lock during global lock checking if necessary